lib/themeroller.js

```diff
@@ -105,7 +105,7 @@ function ThemeRoller( baseThemeCss, vars, options ) {
 
 		// For version < 1.10.0, opacity (w3c) and filter (IE) are combined into the same line.
 		opacityFix = function( opacity ) {
-			return /* w3c */ ( opacity / 100 ).toString().replace( /^0\./, "." ) + /* IE */ ";filter:Alpha(Opacity=" + opacity + ")";
+			return /* w3c */ ( opacity / 100 ).toString().replace( /^0\./, "." ) + /* IE */ ';filter: "alpha(opacity=' + opacity + ')"';
```

Comment:
`filter` is an older property, `-ms-filter` is a newer one. MS introduced `-ms-filter` in IE 8 as `filter` parsing was not compatible with the CSS standard. To make it compatible, the `-ms-filter` value is supposed to be enclosed in quotes (single or double). A difference can be seen at a glance:
```css
filter: alpha(opacity=40);
-ms-filter: "alpha(opacity=40)";
```

So you also need to change `filter` to `-ms-filter`, otherwise it won't work.

Reply:
More info at an old MS blog:
https://web.archive.org/web/20100829182818/http://blogs.msdn.com/b/ie/archive/2008/09/08/microsoft-css-vendor-extensions.aspx
See the "Changes in the filter property syntax" section.

Reply:
Also, we changed `filter` to `-ms-filter` in UI 1.13.0. We shouldn't change what we generate for older jQuery UI versions. You will need to add another `else if` branch here.

Reply:
You mean something like this?
```

	// Opacity fix
	// TODO: Remove `filter` style when dropping support for IE8 and earlier.
	vars.opacityOverlayPerc = vars.opacityOverlay;
	vars.opacityShadowPerc = vars.opacityShadow;
	if ( options.version && semver.lt( options.version, "1.10.0" ) ) {

		// For version < 1.10.0, opacity (w3c) and filter (IE) are combined into the same line.
		opacityFix = function( opacity ) {
			return /* w3c */ ( opacity / 100 ).toString().replace( /^0\./, "." ) + /* IE */ ";filter:Alpha(Opacity=" + opacity + ")";
		};
		vars.opacityOverlay = opacityFix( vars.opacityOverlay );
		vars.opacityShadow = opacityFix( vars.opacityShadow );

	} else if ( options.version && semver.lt( options.version, "1.13.0" ) ) {

		// For version >= 1.10.0 and < 1.13.0, filter has its own separate line and variable name.
		opacityFix = function( opacity ) {
			return ( opacity / 100 ).toString().replace( /^0\./, "." );
		};
		opacityFilter = function( opacity ) {
			return "Alpha(Opacity=" + opacity + ")";
		};
		vars.opacityFilterOverlay = opacityFilter( vars.opacityOverlay );
		vars.opacityFilterShadow = opacityFilter( vars.opacityShadow );
		vars.opacityOverlay = opacityFix( vars.opacityOverlay );
		vars.opacityShadow = opacityFix( vars.opacityShadow );
	} else {

		// For version >= 1.13.0, -ms-filter has its own separate line and variable name.
		opacityFix = function( opacity ) {
			return /* w3c */ ( opacity / 100 ).toString().replace( /^0\./, "." ) + /* IE */ ';-ms-filter: "alpha(opacity=' + opacity + ')"';
		};
		opacityFilter = function( opacity ) {
			return '"alpha(opacity=' + opacity + ')"';
		};
		vars.opacityOverlay = opacityFix( vars.opacityOverlay );
		vars.opacityShadow = opacityFix( vars.opacityShadow );
	}
```

Reply:
Yes for a separate block but the code for `1.13` should be based on the code for `1.12` and not on the code for `<1.10`. I've just realized you've been tweaking the code for the oldest jQuery UI versions here.